src/test/java/org/kohsuke/groovy/sandbox/SandboxTransformerTest.java

```diff
@@ -1103,4 +1103,32 @@ public void sandboxSupportsConstructorsWithVarArgs() throws Exception {
                 "ArrayList.add(null)");
     }
 
+    @Test
+    public void sandboxSupportsReturnStatementsInClosures() throws Exception {
+        assertIntercept(
+                "@groovy.transform.Field\n" +
+                "private static final field = [\n" +
+                "  key: { x -> return x == 123 }\n" +
+                "]\n" +
+                "field.key(123)\n",
+                true,
+                "Script1.field",
+                "LinkedHashMap.key(Integer)",
+                "Integer.compareTo(Integer)");
+        assertIntercept(
+                "File method() {\n" +
+                "  def field = [\n" +
+                "    key: { x -> return x }\n" +
+                "  ]\n" +
+                "  result = field.key(['secret.key'])\n" +
+                "  null\n" +
+                "}\n" +
+                "method()\n" +
+                "result",
```

Comment:
This script used to return an instance of `File` (and intercept the `File` constructor call) in the sandbox.